src/arch/src/aarch64/gic/regs.rs

```diff
@@ -1,24 +1,45 @@
 // Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-mod dist_regs;
-mod icc_regs;
-mod redist_regs;
-
-use crate::aarch64::gic::{Error, Result};
-use kvm_bindings::kvm_device_attr;
-use kvm_ioctls::DeviceFd;
 use std::fmt::Debug;
 use std::iter::StepBy;
 use std::ops::Range;
+
+use kvm_bindings::kvm_device_attr;
+use kvm_ioctls::DeviceFd;
 use versionize::{VersionMap, Versionize, VersionizeResult};
 use versionize_derive::Versionize;
 
+use crate::aarch64::gic::{Error, Result};
+
 #[derive(Debug)]
 pub struct GicRegState<T: Versionize> {
     pub(crate) chunks: Vec<T>,
 }
 
+/// Structure for serializing the state of the Vgic ICC regs
+#[derive(Debug, Default, Versionize)]
+pub struct VgicSysRegsState {
+    pub main_icc_regs: Vec<GicRegState<u64>>,
+    pub ap_icc_regs: Vec<Option<GicRegState<u64>>>,
+}
+
+/// Structure used for serializing the state of the GIC registers
+#[derive(Debug, Default, Versionize)]
+pub struct GicState {
+    /// Temp doc
+    pub dist: Vec<GicRegState<u32>>,
+    /// Temp doc
+    pub gic_vcpu_states: Vec<GicVcpuState>,
+}
+
+/// Structure used for serializing the state of the GIC registers for a specific vCPU
+#[derive(Debug, Default, Versionize)]
+pub struct GicVcpuState {
+    pub rdist: Vec<GicRegState<u32>>,
```

Comment:
This struct is also gicv3 specific, since there is no redistributors for v2

Reply:
Can we write traits for these structs and define them for each GIC version?

Reply:
From the top of my head: one idea that might work would be to use an enum for each state. This would require some more translation code as we would need to change the structure of the snapshot.

Reply:
@dianpopa and I talked offline and this is indeed an option that she brought up. I am fine however with keeping things as they are, since I don't see that much benefit apart from some slightly increased code expressiveness, wdyt?

Reply:
I did take into account using an [enum for each state](https://github.com/dianpopa/firecracker/tree/gicv2_enum), however as decided to keep the current version since it does not require a snapshot version bump for gicv3 and so there is no hidden cost with it. 
One practical reason for using an enum could be to make the transition smoother to GICv4 (in case we want to support that in the future). However, gicv4 has the same programmer model as Gicv3 so we would be fine with using the same GIcState. 
Additionally, the enum implementation can be done incrementally on the current one.
So @sandreim are you ok with merging the PR like it is now or you foresee problems with current implementation?

Reply:
I think we are fine with less changes in state for now, unless we have clear data that shows otherwise (Gicv4 doesn't).